crates/cli/src/cli/generate.rs

```diff
@@ -0,0 +1,213 @@
+use std::path::PathBuf;
+use std::io::{self, Read};
+use std::fs;
+use clap::Args;
+use serde::Serialize;
+use alloy::primitives::hex;
+
+use crate::error::Error;
+use crate::meta::types::dotrain::source_v1::DotrainSourceV1;
+use crate::meta::RainMetaDocumentV1Item;
+use crate::meta::KnownMagic;
+
+/// Generate deployment data for DotrainSourceV1 content
+#[derive(Args)]
+pub struct Generate {
+    /// Path to input .rain file. If not provided, reads from stdin
+    #[arg(short = 'i', long = "input-path")]
+    input_path: Option<PathBuf>,
+
+    /// Path to output JSON file. If not provided, prints to stdout  
+    #[arg(short = 'o', long = "output-path")]
+    output_path: Option<PathBuf>,
+}
+
+/// Deployment data for publishing to MetaBoard
+#[derive(Debug, Clone, Serialize, serde::Deserialize)]
+pub struct DeploymentData {
+    /// Keccak256 hash of the content (hex-encoded with 0x prefix)
+    pub subject: String,
+    /// CBOR-encoded metadata bytes (hex-encoded with 0x prefix)
+    pub meta_bytes: String,
+    /// Complete calldata for MetaBoard.emitMeta() (hex-encoded with 0x prefix)
+    pub calldata: String,
+}
+
+/// Validate dotrain content - basic checks
+fn validate_dotrain_content(content: &str) -> Result<(), Error> {
+    // Check if content is empty or only whitespace
+    if content.trim().is_empty() {
+        return Err(Error::InvalidInput(
+            "Dotrain content cannot be empty".to_string(),
+        ));
+    }
+
+    // Try to create DotrainSourceV1 to ensure it's valid
+    let _dotrain_source = DotrainSourceV1(content.to_string());
+
+    Ok(())
+}
+
+/// Generate deployment data for DotrainSourceV1 content
+/// Validates content and returns subject hash, meta bytes, and calldata
+fn generate_dotrain_deployment(content: &str) -> Result<DeploymentData, Error> {
+    // Validate content
+    validate_dotrain_content(content)?;
```

Comment:
i think the validation logic can be expanded here instead of having it as a separate function